openedx/core/djangoapps/content/block_structure/tasks.py

```diff
@@ -15,9 +15,12 @@
     default_retry_delay=settings.BLOCK_STRUCTURES_SETTINGS['BLOCK_STRUCTURES_TASK_DEFAULT_RETRY_DELAY'],
     max_retries=settings.BLOCK_STRUCTURES_SETTINGS['BLOCK_STRUCTURES_TASK_MAX_RETRIES'],
 )
-def update_course_in_cache(course_key):
+def update_course_in_cache(course_id):
     """
     Updates the course blocks (in the database) for the specified course.
     """
-    course_key = CourseKey.from_string(course_key)
-    api.update_course_in_cache(course_key)
+    try:
+        course_key = CourseKey.from_string(course_id)
+        api.update_course_in_cache(course_key)
+    except Exception as exc:   # pylint: disable=broad-except
```

Comment:
Thoughts on how to make this a more specific exception, or is the broad except the way to go here?